packages/sanity/src/core/util/userHasRole.ts

```diff
@@ -0,0 +1,24 @@
+import type {CurrentUser} from '@sanity/types'
+
+/**
+ * Checks whether or not the given user has the role with the given ID
+ *
+ * @param user - The user to check (currently only the current user is supported)
+ * @param roleId - The ID of the role to check for
+ * @returns True if the user has the role, false otherwise
+ * @example
+ * Fetch the current user and check if they have the role "administrator":
+ * ```
+ * import {userHasRole, useCurrentUser} from 'sanity'
+ *
+ * export function MyComponent() {
+ *   const user = useCurrentUser()
+ *   const hasAdminRole = userHasRole(user, 'administrator')
+ *   return <div>Is administrator: {hasAdminRole ? 'Yes' : 'No'}</div>
+ * }
+ * ```
+ * @public
+ */
+export function userHasRole(user: CurrentUser, roleId: string): boolean {
```

Comment:
Nice! 👍 

Maybe it would be nice to have support for an array of roles as well?

Something like:

```ts
export function userHasRole(user: CurrentUser, roleId: string | string[]): boolean {
  const {roles} = user

  if (Array.isArray(roleId)) {
    return roles.filter((r) => roleId.includes(r.name)).length === roleId.length
  }

  return roles.some((role) => role.name === roleId)
}
```

```ts
const hasRoles = userHasRole(user, ['developer', 'custom-role', 'another-role'])
```

Reply:
It is unclear to me in the array form (without reading the code) whether or not it is "the user needs all of these roles", or if it is "the user needs _one_ of these roles". Maybe it would be better to have separate methods for this use case, which could have more descriptive names?

Reply:
Will merge as-is for now and we can consider expanding the API since it is not a breaking change.